sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -2309,8 +2355,22 @@ CurlConnection::CurlConnection(
         _detail::DefaultFailedToGetNewConnectionTemplate + hostDisplayName + ". "
         + std::string("curl_easy_init returned Null"));
   }
+  
   CURLcode result;
 
+  if (request.GetHeader("connection").HasValue() && request.GetHeader("keep-alive").HasValue())
+  {
+    auto connectionHeader = Azure::Core::_internal::StringExtensions::ToLower(
+        request.GetHeader("connection").Value());
+    auto keepAliveHeader = Azure::Core::_internal::StringExtensions::ToLower(
+        request.GetHeader("keep-alive").Value());
+
+    if (connectionHeader == "keep-alive")
```

Comment:
If the new response doesn't contain a keep-alive header, should you reset the keep-alive state to None to make sure that you don't have a stale keep-alive value?

Reply:
done